feat: add C implementation for `stats/base/dists/planck/mode`

yuvimittal opened the pull request:
Resolves #4910 

## Description

> What is the purpose of this pull request?

This pull request:

-  Add C implementation for @stdlib/stats/base/dists/planck/mode 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #4910 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

stdlib-bot committed:
docs: update related packages sections

PR-URL: https://github.com/stdlib-js/stdlib/pull/4784
Reviewed-by: Athan Reines <kgryte@gmail.com>

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
Add C implementation for @stdlib/stats/base/dists/planck/mode

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
Add C implementation for @stdlib/stats/base/dists/planck/mode

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Neerajpathak07 commented on lib/node_modules/@stdlib/constants/float32/pi/README.md:
```diff
@@ -108,6 +108,14 @@ Macro for the mathematical constant [π][pi].
 
+* * *
+
+## See Also
```
same comment for this you can create a new PR anytime to address these changes.

Neerajpathak07 commented on lib/node_modules/@stdlib/blas/ext/base/sapxsumkbn/README.md:
```diff
@@ -166,7 +166,7 @@ console.log( v );
 
 -   <span class="package-name">[`@stdlib/blas/ext/base/dapxsumkbn`][@stdlib/blas/ext/base/dapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a constant to each double-precision floating-point strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
--   <span class="package-name">[`@stdlib/blas/ext/base/gapxsumkbn`][@stdlib/blas/ext/base/gapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a constant to each strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
+-   <span class="package-name">[`@stdlib/blas/ext/base/gapxsumkbn`][@stdlib/blas/ext/base/gapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a scalar constant to each strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
```
I don't think there is a need to make changes to the the `blas/ext/base` packages as we should adhere to the purpose of the PR.

yuvi-mittal replied:
As I told you earlier @Neerajpathak07 , i did not make these changes . They got pushed while i was pushing my changes , see the commits and you will understand what i am saying.

Neerajpathak07 replied:
Np, you can rebase the commit and fix it. Well I didn't know if you were referring to this PR in particular which is why thought of having an heads up. My apologies if it sounded to pushy!!

yuvimittal committed:
changed file back to normal

Signed-off-by: yuvi-mittal <128018763+yuvi-mittal@users.noreply.github.com>

yuvimittal committed:
removed unnecccesary file

Signed-off-by: yuvi-mittal <128018763+yuvi-mittal@users.noreply.github.com>

yuvimittal committed:
updated doc

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
updated doc

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
fix: remove unused dependencies from `planck/mode`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/src/main.c:
```diff
@@ -0,0 +1,37 @@
+* Returns the mode of a Planck distribution with shape parameter `lambda`.
+*
+* @param lambda   shape parameter
+* @return         mode or NaN if invalid input
```
```suggestion
* @return         mode
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/package.json:
```diff
@@ -14,19 +14,22 @@
   "repository": {
-    "type": "git",
-    "url": "git://github.com/stdlib-js/stdlib.git"
+      "type": "git",
+      "url": "git://github.com/stdlib-js/stdlib.git"
```
```suggestion
    "type": "git",
    "url": "git://github.com/stdlib-js/stdlib.git"
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/src/addon.c:
```diff
@@ -0,0 +1,23 @@
+#include "stdlib/stats/base/dists/planck/mode.h"
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
```
```suggestion
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/lib/native.js:
```diff
@@ -0,0 +1,58 @@
+// MAIN //
+
+/**
+* Returns the mode for Planck distribution with shape parameter `lambda`.
```
```suggestion
* Returns the mode for a Planck distribution with shape parameter `lambda`.
```

Planeshifter commented on lib/node_modules/@stdlib/constants/float32/pi/README.md:
```diff
@@ -108,6 +108,14 @@ Macro for the mathematical constant [π][pi].
+-   <span class="package-name">[`@stdlib/constants/float32/fourth-pi`][@stdlib/constants/float32/fourth-pi]</span><span class="delimiter">: </span><span class="description">1/4 times π.</span>
+-   <span class="package-name">[`@stdlib/constants/float32/half-pi`][@stdlib/constants/float32/half-pi]</span><span class="delimiter">: </span><span class="description">1/2 times π.</span>
+-   <span class="package-name">[`@stdlib/constants/float32/two-pi`][@stdlib/constants/float32/two-pi]</span><span class="delimiter">: </span><span class="description">2π.</span>
+
```
```suggestion
```

Planeshifter commented on lib/node_modules/@stdlib/blas/ext/base/sapxsumkbn/README.md:
```diff
@@ -166,7 +166,7 @@ console.log( v );
 
 -   <span class="package-name">[`@stdlib/blas/ext/base/dapxsumkbn`][@stdlib/blas/ext/base/dapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a constant to each double-precision floating-point strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
--   <span class="package-name">[`@stdlib/blas/ext/base/gapxsumkbn`][@stdlib/blas/ext/base/gapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a constant to each strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
+-   <span class="package-name">[`@stdlib/blas/ext/base/gapxsumkbn`][@stdlib/blas/ext/base/gapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a scalar constant to each strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
```
```suggestion
-   <span class="package-name">[`@stdlib/blas/ext/base/gapxsumkbn`][@stdlib/blas/ext/base/gapxsumkbn]</span><span class="delimiter">: </span><span class="description">add a constant to each strided array element and compute the sum using an improved Kahan–Babuška algorithm.</span>
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/README.md:
```diff
@@ -118,6 +118,96 @@ for ( i = 0; i < lambda.length; i++ ) {
+
+#### stdlib_base_dists_planck_mode( lambda )
+
+Evaluates the mode for planck distribution.
```
```suggestion
Returns the [mode][mode] of a Planck distribution with shape parameter `lambda`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/README.md:
```diff
@@ -118,6 +118,96 @@ for ( i = 0; i < lambda.length; i++ ) {
+
+The function accepts the following arguments:
+
+-   **lambda**: `[in] double` shape parameter
```
```suggestion
-   **lambda**: `[in] double` shape parameter.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/README.md:
```diff
@@ -118,6 +118,96 @@ for ( i = 0; i < lambda.length; i++ ) {
+    for ( i = 0; i < 25; i++ ) {
+        lambda = random_uniform( 0.1, 10.0 );
+        y = stdlib_base_dists_planck_mode( lambda );
+        printf( "lambda: %lf, E(X;λ): %lf\n", lambda, y );
```
```suggestion
        printf( "lambda: %lf, mode(X;λ): %lf\n", lambda, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/examples/c/example.c:
```diff
@@ -0,0 +1,37 @@
+	for ( i = 0; i < 25; i++ ) {
+		lambda = random_uniform( 1.0, 10.0 );
+		y = stdlib_base_dists_planck_mode( lambda );
+		printf( "λ: %lf, F(x;λ): %lf\n", lambda , y );
```
```suggestion
		printf( "λ: %lf, mode(X;λ): %lf\n", lambda , y );
```

Planeshifter committed:
style: fix spacing in C benchmark for `planck/mode`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/constants/float32/pi/README.md:
```diff
@@ -118,6 +118,16 @@ Macro for the mathematical constant [π][pi].
 
 [pi]: https://en.wikipedia.org/wiki/Pi
 
+<!-- <related-links> -->
+
+[@stdlib/constants/float32/fourth-pi]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/constants/float32/fourth-pi
+
+[@stdlib/constants/float32/half-pi]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/constants/float32/half-pi
+
+[@stdlib/constants/float32/two-pi]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/constants/float32/two-pi
+
+<!-- </related-links> -->
+
```
```suggestion
```

Planeshifter committed:
bench: revert changes

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		lambda[i] = ( random_uniform( 0.0, 10.0 ) + 1.0 );
```
```suggestion
		lambda[ i ] = random_uniform( 1.0, 10.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		y = stdlib_base_dists_planck_mode( lambda[i % 100] );
```
```suggestion
		y = stdlib_base_dists_planck_mode( lambda[ i % 100 ] );
```

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (dismissed):
Should be ready to land after some edits. Thanks, @yuvi-mittal!

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double lambda[100];
```
```suggestion
	double lambda[ 100 ];
```

kgryte committed:
style: add spaces

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/src/main.c:
```diff
@@ -0,0 +1,37 @@
+*
+* @example
+* double y = stdlib_base_dists_planck_mode( 0.1 );
+* // returns 0
```
```suggestion
* // returns 0.0
```

kgryte committed:
docs: fix missing decimal

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,67 @@
+	len = 100;
+	lambda = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		lambda[ i ] = uniform( 0.1, 11 );
```
```suggestion
		lambda[ i ] = uniform( 0.1, 11.0 );
```

kgryte committed:
style: fix missing decimal

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/test/test.native.js:
```diff
@@ -0,0 +1,77 @@
+});
+
+tape( 'if provided `NaN` for `lambda`, the function returns `NaN`', opts, function test( t ) {
+	var v;
+	v = mode( NaN );
```
```suggestion
	var v = mode( NaN );
```

kgryte committed:
style: combine declaration and assignment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/README.md:
````diff
@@ -118,6 +118,96 @@ for ( i = 0; i < lambda.length; i++ ) {
+
+```c
+double out = stdlib_base_dists_planck_mode( 0.1 );
+// returns ~0
````
```suggestion
// returns 0.0
```

kgryte committed:
style: fix return annotation

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: fix missing decimals

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (dismissed):
LGTM pending passing CI.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/planck-mode

kgryte commented:
@Planeshifter FYI: this PR is a duplicate of https://github.com/stdlib-js/stdlib/pull/5695 by the same author which you already attempted to edit and review. The CI failures happened in that PR, as well.

Always worthwhile navigating to the issue a PR attempts to close in order to check the status of potential duplicates prior to beginning PR review.

kgryte commented:
Should also cross-check that both PRs are consistent in the changes made.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/binding.gyp:
```diff
@@ -0,0 +1,170 @@
+{
+  # List of files to include in this file:
+  'includes': [
+    './include.gypi ',
```
@Planeshifter For whatever reason, it seems that the author added a space here.

```suggestion
    './include.gypi',
```

kgryte committed:
build: fix file path

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/include.gypi:
```diff
@@ -0,0 +1,53 @@
+  # Define variables to be used throughout the configuration for all targets:
+  'variables': {
+    # Source directory:
+    'src_dir': ' ./src',
```
```suggestion
    'src_dir': './src',
```

kgryte committed:
build: fix path

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
@Planeshifter Would you mind cross checking the `*.gyp` files with existing files? For whatever reason, the author made various unnecessary changes instead of just copy-pasting.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/src/Makefile:
```diff
@@ -0,0 +1,70 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
```
```suggestion
#    http://www.apache.org/licenses/LICENSE-2.0
```

kgryte committed:
docs: fix missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/benchmark/c/Makefile:
```diff
@@ -0,0 +1,146 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License .
```
```suggestion
# limitations under the License.
```

kgryte committed:
docs: remove space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/examples/c/Makefile:
```diff
@@ -0,0 +1,146 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License .
```
```suggestion
# limitations under the License.
```

kgryte committed:
docs: fix space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/lib/native.js:
```diff
@@ -0,0 +1,58 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License .
```
```suggestion
* limitations under the License.
```

kgryte committed:
docs: remove space

Signed-off-by: Athan <kgryte@gmail.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mode/README.md:
```diff
@@ -118,6 +118,96 @@ for ( i = 0; i < lambda.length; i++ ) {
+    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
+    return min + ( v * (max - min) );
+}
+int main( void ) {
```
```suggestion
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v * (max - min) );
}

int main( void ) {
```

Planeshifter committed:
style: add empty line

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
The PR is now ready to land after addressing several issues. Moving forward, please take extra care to review for any errors in future submissions, @yuvi-mittal. Your attention to detail is greatly appreciated! Thanks for your contribution!
